salt/modules/nilrt_ip.py

```diff
@@ -515,10 +631,40 @@ def _persist_config(section, token, value):
         raise salt.exceptions.CommandExecutionError(exc_msg)
 
 
+def set_ethercat(interface, master_id):
+    '''
+    Configure specified adapter to use EtherCAT adapter mode. If successful, the target will need reboot if it doesn't
+    already use EtherCAT adapter mode, otherwise will return true.
+
+    :param interface: interface label
+    :param master_id: EtherCAT Master ID
+    :return: True if the settings were applied, otherwise an exception will be thrown.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' ip.set_ethercat interface-label master-id
+    '''
+    if __grains__['lsb_distrib_id'] == 'nilrt':
+        initial_mode = _get_adaptermode_info(interface)
+        _set_adapter_mode(interface, 'EtherCAT')
```

Comment:
why you spell 'EtherCAT' here, but do this in other places: if initial_mode == 'ethercat':

Reply:
I will make a constant:
```python
NIRTCFG_ETHERCAT = 'EtherCAT'
```